### DISCUSSION TITLE
tlscrypt v2 support

### DISCUSSION THREAD

**Discussion by tkmw1985** [pivpn/pivpn#1379] [Feature requests]
Hello !  

Is it possible to add tlscrypt v2 in this super awsome script ?

**Comment by TinCanTech** [comment]
If you are interested in TLS-Crypt-V2 you may like to try this: https://github.com/TinCanTech/easy-tls

**Comment by tkmw1985** [comment]
Hello dear TinCanTech !

Yes i know already about it but i do not know how to do it properly can not find step by step command or manual for understand ! 

Will be much appreciated if you point me. 

Thank you.

**Comment by TinCanTech** [comment]
There are 24 pages of help in the wiki.

If you would like to open an issue [here](https://github.com/TinCanTech/easy-tls/issues), I'll see what I can do.

**Comment by Hylosium** [comment]
Good noon,

I am trying to use easy-tls with pivpn but I have OpenVPN 2.4 installed and when I try to update to 2.5 it says that doesn't support armhf (my distribution is armbian for orangepilite, based on Ubuntu Focal, it will attach photos)
![Captura de pantalla de 2021-10-01 22-03-28](https://user-images.githubusercontent.com/90805628/135680266-b7ac456a-b2d2-48b2-8506-ab59ec6e9f0a.png) ![Captura de pantalla de 2021-10-01 22-10-12](https://user-images.githubusercontent.com/90805628/135680540-c9c980c4-b0b9-4f34-87fb-d2fa203dde53.png)

What could I do to solve this error and continuing installing [05 = Quick Start Normal mode](https://github.com/TinCanTech/easy-tls/wiki/05-=-Quick-Start-Normal-mode) I think this is what I need to use tls-crypt v2 and bypass DPI.

The code I use is:
`sudo -s`
`wget -O - https://swupdate.openvpn.net/repos/repo-public.gpg | apt-key add -`
`echo "deb http://build.openvpn.net/debian/openvpn/stable focal main" > /etc/apt/sources.list.d/openvpn-aptrepo.list`
`exit`

Thanks in advance.
 -------
Edit, I found the 2.5 OpenVPN version for armhf [Link for Dowload](https://packages.debian.org/sid/armhf/openvpn/download)
Now that I have to follow [05 = Quick Start Normal mode](https://github.com/TinCanTech/easy-tls/wiki/05-=-Quick-Start-Normal-mode), do I need to have easyrsa3 or with easyrsa would it be possible?

**Reply by Hylosium** [reply]
Only as update on 15/11/2021, I made it work, but I installed all the packages one by one but without using pivpn, thats not what OP wanted, but with easy-tls, openvpn for arm and easy-rsa I made it work. I'm using tls-crypt-v2, was a bit difficult but I could do it.
Thanks @TinCanTech.

**Reply by TinCanTech** [reply]
If you need help with Easy-TLS then please post me an issue.

**Reply by island3r** [reply]
> Only as update on 15/11/2021, I made it work, but I installed all the packages one by one but without using pivpn, thats not what OP wanted, but with easy-tls, openvpn for arm and easy-rsa I made it work. I'm using tls-crypt-v2, was a bit difficult but I could do it. Thanks @TinCanTech.

So any chance of getting some guidance? I managed to install pivpn with openvpn 2.5, but after that it gets a bit complicated.

**Comment by coolapso** [comment]
Moving thread to discussions.

**Reply by TinCanTech** [reply]
@4s3ti Thanks for making this a discussion.

For anybody wanting to use Easy-TLS to build TLS-Crypt-V2 keys and Inline files for your VPN, here are some pages to help you get started.

This [cheatSheet](https://github.com/TinCanTech/easy-tls/wiki/cheatSheet) shows how to use TLS-Crypt-V2 GROUP keys, which is the simplest way to start.  You Need one Server key and one Client key (Client key can be shared by numerous clients). This can be easily expanded as you enrol more clients (And even servers).

This [guide](https://github.com/TinCanTech/easy-tls/wiki/Quick-Start) will take you through a Full Easy-TLS setup.

If you have difficulty using Easy-TLS then please start a discussion here:<br>https://github.com/TinCanTech/easy-tls/discussions/categories/easytls-and-related-help

If you prefer, you can leave feed-back here in this **pivpn** discussion.

I apologies in advance for errors in the wiki; It takes time to prepare documentation without help.

**Reply by island3r** [reply]
@TinCanTech Thank you for your excellent work. I decided against creating a new thread because I  thought it would be pretty arbitrary. So I decided to follow your instructions [here](https://forums.openvpn.net/viewtopic.php?f=4&t=30111#p93931) on my test mule pi. However it complains that it can't find client-name.crt in the pki/issued directory. Isn't this supposed to be generated automatically?

**Reply by TinCanTech** [reply]
> However it complains that it can't find client-name.crt in the pki/issued directory. Isn't this supposed to be generated automatically?

Easy-TLS does not generate X509 certificates.

In this case **pivpn** would generate your X509(Easy-RSA)PKI and `client-name.crt`.

**Reply by tkmw1985** [reply]
Work like a charm with pivpn !

**Comment by orazioedoardo** [comment]
This is implemented now: https://github.com/pivpn/pivpn/releases/tag/v4.11.0
